gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystem.java

```diff
@@ -1072,20 +1085,14 @@ private GoogleCloudStorageItemInfo getFileInfoInternal(
 
     ExecutorService dirExecutor =
         options.isStatusParallelEnabled()
-            ? resourceId.isDirectory()
-                ? Executors.newSingleThreadExecutor(DAEMON_THREAD_FACTORY)
-                : Executors.newFixedThreadPool(2, DAEMON_THREAD_FACTORY)
+            ? newSingleThreadExecutor(DAEMON_THREAD_FACTORY)
```

Comment:
Would changing dir listing from 2 threads to 1 not affect the performance?

Reply:
It reduces number of async requests from 2 to 1, that's why thread pool size also reduced to 1 thread.